rom/dev/tests/rom_integration_tests/test_idevid_derivation.rs

```diff
@@ -54,3 +67,48 @@ fn test_idev_subj_key_id_algo() {
             .unwrap();
     }
 }
+
+fn fuses_with_random_uds() -> Fuses {
+    const UDS_LEN: usize = core::mem::size_of::<u32>() * 12;
+    let mut uds_bytes = [0; UDS_LEN];
+    rand_bytes(&mut uds_bytes).unwrap();
+    let mut uds_seed = [0u32; 12];
+
+    for (word, bytes) in uds_seed.iter_mut().zip(uds_bytes.chunks_exact(4)) {
+        *word = u32::from_be_bytes(bytes.try_into().unwrap());
+    }
+    Fuses {
+        uds_seed,
+        ..Default::default()
+    }
+}
+
+#[test]
+#[cfg_attr(not(feature = "slow_tests"), ignore)]
+fn test_generate_csr_stress() {
+    const NUM_TESTS: usize = 1000;
+
+    for _ in 0..NUM_TESTS {
+        let fuses = fuses_with_random_uds();
+        let (mut hw, image_bundle) =
+            helpers::build_hw_model_and_image_bundle(fuses, ImageOptions::default());
+
+        let Csrs {
+            uploaded,
+            downloaded: _,
+        } = generate_csr(&mut hw, &image_bundle);
+
+        // Ensure CSR is valid X.509
+        X509Req::from_der(&uploaded).unwrap_or_else(|_| {
```

Comment:
Can we also check that the signature verifies with the public key? I think it should be something like 

```rust
let pub = req.public_key().unwrap();
assert!(req.verify(pub).unwrap());
```

Reply:
Thanks! I added the check.